API/src/app.py

```diff
@@ -0,0 +1,132 @@
+# API/src/app.py
+
+import sys
+import os
+import threading
+import time
+import logging
+from flask import Flask, render_template, request, jsonify
+from pyxmas import (
+    get_ttn_access_layer,
+    SensorsTTNPayload,
+    ActionsTTNPayload,
+    ActionsEnum,
+    sensors_topic,
+    actions_topic,
+)
+from pyxmas.settings import TTN_APP_ID, TTN_API_KEY, TTN_BASE_URL, TTN_PORT
+from datetime import datetime, timezone
+
+# Ensure pyxmas is in the Python path
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'pyxmas')))
+
+app = Flask(__name__)
+app.config['SECRET_KEY'] = 'your_secret_key'  # Replace with a secure key in production
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# Global variable to store the latest sensor data
+latest_sensor_data = {
+    'brightness': 'N/A',
+    'loudness': 'N/A',
+    'temperature': 'N/A',
+    'timestamp': 'N/A'
+}
+data_lock = threading.Lock()
+
+def on_sensors_data(sensors_data: SensorsTTNPayload):
+    global latest_sensor_data
```

Comment:
From previous comment: logic_map already has this, and you should avoid using the keyword `global` with python